src/Pos/Wallet/Web/Tracking.hs

```diff
@@ -192,8 +200,40 @@ syncWalletsWithGState
       Ether.MonadReader' GenesisUtxo m)
     => [EncryptedSecretKey]
     -> m ()
-syncWalletsWithGState encSKs = withBlkSemaphore_ $ \tip ->
-    tip <$ mapM_ (syncWalletWithGStateUnsafe @ssc) encSKs
+syncWalletsWithGState encSKs = forM_ encSKs $ \encSK -> do
+    let wAddr = encToCId encSK
+    whenJustM (WS.getWalletSyncTip wAddr) $ \wTip ->
+        if wTip == genesisHash then syncDo encSK Nothing
+        else DB.blkGetHeader wTip >>= \case
+            Nothing ->
+                logWarning $
+                sformat ("Couldn't get block header of wallet "%build
+                            %" by last synced hh: "%build) wAddr wTip
```

Comment:
Such error seems somewhat fatal. Isn't it better to actually throw an error here?